frameworks/kafka/setup-helper/setup_helper.go

```diff
@@ -162,14 +163,26 @@ func setListeners() error {
 	} else { // No TLS, no Kerberos, Plaintext only
 		listeners = append(listeners,
 			getListener("PLAINTEXT", brokerPort))
+		// NOTE: To be consistent with the legacy behavior of the 2.0.X Kafka series,
+		// we advertise the IP address rather than the host name.
 		advertisedListeners = append(advertisedListeners,
-			getAdvertisedListener("PLAINTEXT", brokerPort))
+			getListener("PLAINTEXT", brokerPort))
 	}
 
 	err := writeToWorkingDirectory(listenersProperty,
 		"listeners="+strings.Join(listeners, ","))
-	err = writeToWorkingDirectory(advertisedListenersProperty,
-		"advertised.listeners="+strings.Join(advertisedListeners, ","))
+
+	// NOTE: To be consistent with the legacy behavior of the 2.0.X Kafka series,
+	// when there is no security enabled, we must honor the kafka.kafka_advertise_host_ip
+	// configuration parameter
+	if !kerberosEnabled && !tlsEncryptionEnabled && !getBooleanEnvvar(advertiseHostIPEnvvar) {
+		// Write an empty file.
+		err = writeToWorkingDirectory(advertisedListenersProperty, "")
```

Comment:
Isn't `advertisedListeners` empty in this case anyway? So we just need to ensure that we don't write `advertised.listeners=` if this is the case?

Reply:
I'm just ensuring that the file is there so that `cat` can't possibly fail in the svc.yml.

Reply:
(the question was more as to whether we could set the condition on the length of the advertised listeners instead of checking the multi-term boolean expression)